imap_processing/tests/test_cli.py

```diff
@@ -316,6 +320,54 @@ def test_hit_l1a(mock_hit_l1a, mock_instrument_dependencies):
     assert mocks["mock_upload"].call_count == 2
 
 
+@pytest.mark.usefixtures("_unset_metakernel_path")
+def test_pre_processing(spice_test_data_path):
+    """Test coverage for ProcessInstrument.pre_processing method()."""
+    kernels_to_furnish = ["naif0012.tls", "imap_sclk_0000.tsc"]
+    dependency_str = (
+        "["
+        '{"type": "science","files": ["imap_hi_l2a_sensor45-de_20100105_v001.cdf"]},'
+        f'{{"type":"spice","files": ["{'", "'.join(kernels_to_furnish)}"]}},'
+        '{"type": "spice","files": ["imap_2010_104_01.repoint.csv"]}'
+        "]"
+    )
+
+    def download_side_effect(path: Path):
+        """Copy kernels from spice_test_data_path to expected DATA_DIR location."""
+        if (spice_test_data_path / path.name).exists():
+            path.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copy2(spice_test_data_path / path.name, path)
+
+    # Test that the expected kernels are furnished within the function that
+    # gets substituted in for the processing function.
+    def do_processing_side_effect(*args, **kwargs):
+        """Check that the expected kernels are furnished"""
+        kernel_count = spiceypy.ktotal("ALL")
+        furnished_kernels = [
+            Path(spiceypy.kdata(i, "ALL")[0]).name for i in range(kernel_count)
+        ]
+        np.testing.assert_array_equal(
+            sorted(furnished_kernels), sorted(kernels_to_furnish)
+        )
+        return [xr.Dataset()]
+
+    with (
+        mock.patch("imap_data_access.processing_input.download") as mock_download,
+        mock.patch("imap_processing.cli.load_cdf"),
+        mock.patch("imap_processing.cli.Hi.do_processing") as mock_do_processing,
+        mock.patch("imap_processing.cli.ProcessInstrument.post_processing"),
+    ):
+        mock_download.side_effect = download_side_effect
+        mock_do_processing.side_effect = do_processing_side_effect
+
+        instrument = Hi(
+            "l1b", "sensor45-de", dependency_str, "20100105", None, "v001", True
+        )
+        # Verify no kernels are furnished prior to calling process
+        assert spiceypy.ktotal("ALL") == 0
+        instrument.process()
```

Comment:
Do you want to use the verification on the number after processing too? Not necessary, but would you want to say anything about what happens if you call `instrument.process()` again or with some kernels already pre-loaded?
```suggestion
        # Verify no kernels are furnished prior to calling process
        assert spiceypy.ktotal("ALL") == 0
        instrument.process()
        assert spiceypy.ktotal("ALL") == len(kernels_to_furnish)
```

Reply:
Great question. I just tested calling `instrument.process()` again and SPICE happily re-furnishes the same files. I hadn't really thought about the desired behavior for that case. It would be easy to clear the furnished kernels prior to the `furnsh()` call to ensure that only what is passed in are used 🤷

Reply:
I decided that it makes sense to clean up the furnished kernels in `post_processing()`. It isn't guaranteed to run but as we discussed probably isn't that important. If we find the need to do it differently, it's easy to change.